TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Robot/Robot.java

```diff
@@ -1,68 +1,62 @@
-import com.qualcomm.robotcore.hardware.Servo;
 
-/**
- * Created by pston on 11/12/2017
- */
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+import org.firstinspires.ftc.teamcode.GMR.Robot.SubSystems.DriveTrain;
+import org.firstinspires.ftc.teamcode.GMR.Robot.SubSystems.RobotLift;
 
 public class Robot {
 
     public DriveTrain driveTrain;
-    public BlockLift blockLift;
-    public RelicGrab relicGrab;
-    public LatchLift latchLift;
+
+    public RobotLift robotLift;
+
 
     private DcMotor leftFront;
-    private DcMotor rightFront;
     private DcMotor leftRear;
+    private DcMotor rightFront;
     private DcMotor rightRear;
-    private NavxMicroNavigationSensor gyroReference;
     private IntegratingGyroscope gyro;
 
     private DcMotor liftMotor;
-    private Servo topLeftGrab;
-    private Servo topRightGrab;
-    private Servo bottomLeftGrab;
-    private Servo bottomRightGrab;
 
-    private DcMotor relicLift;
-    private Servo slideLift;
-    private Servo relicTilt;
-    private Servo relicClamp;
+    private DcMotor armMotor;
+    private int armEncoder;
 
-    private Servo rightColor;
-    private Servo leftColor;
+    public Robot (HardwareMap hardwareMap, Telemetry telemetry){
 
-
-    public Robot (HardwareMap hardwareMap, Telemetry telemetry) {
-
-        gyro = hardwareMap.get(NavxMicroNavigationSensor.class, "navx");
         leftFront = hardwareMap.dcMotor.get("leftfront");
-        rightFront = hardwareMap.dcMotor.get("rightfront");
         leftRear = hardwareMap.dcMotor.get("leftrear");
+        rightFront = hardwareMap.dcMotor.get("rightfront");
         rightRear = hardwareMap.dcMotor.get("rightrear");
+        gyro = hardwareMap.get(NavxMicroNavigationSensor.class, "navx");
 
         liftMotor = hardwareMap.dcMotor.get("liftmotor");
 
-        driveTrain = new DriveTrain(leftFront, rightFront, leftRear, rightRear, gyro, telemetry);
+        armMotor = hardwareMap.dcMotor.get("armmotor");
 
-        blockLift = new BlockLift(liftMotor, topLeftGrab, topRightGrab, bottomLeftGrab, bottomRightGrab);
+        driveTrain = new DriveTrain(leftFront, rightFront, leftRear, rightRear, gyro, telemetry);
 
-        relicGrab = new RelicGrab(relicLift, slideLift, relicTilt, relicClamp);
 
-        blockLift.clamp(false, true, false, false);
+        robotLift = new RobotLift(liftMotor, telemetry);
+    }
 
-        latchLift = new LatchLift(liftMotor, telemetry);
+    public void rake(boolean bumper, float trigger){
+        if(bumper && trigger != 1){
+            armMotor.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+            armMotor.setPower(1.0);
+        } else if (!bumper && trigger == 1){
+            armMotor.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+            armMotor.setPower(-1.0);
+        } else {
+            armEncoder = armMotor.getCurrentPosition();
+            armMotor.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+            armMotor.setTargetPosition(armEncoder);
```

Comment:
Consider: `setPower(0)` might save us from having to change the motor modes repeatedly.